analytic_engine/src/manifest/details.rs

```diff
@@ -317,12 +524,14 @@ impl Manifest for ManifestImpl {
             load_req.table_id,
             self.store.clone(),
         );
-        let snapshotter = Snapshotter {
-            location,
+        let storage_snapshot_builder = StorageSnapshotBuilder {
             log_store,
             snapshot_store,
         };
-        let snapshot = snapshotter.create_latest_snapshot().await?;
+        let snapshotter = Snapshotter {
+            snapshot_builder: storage_snapshot_builder,
+        };
+        let snapshot = snapshotter.snapshot().await?;
```

Comment:
In the previous implementation, only the latest snapshot is fetched but won't be stored and the old logs would also be kept. If so, a snapshot will be generated and stored every time read the table manifest data, and I guess it is unacceptable.